### PR TITLE
Update verify

### PR DESCRIPTION
**Pull request by SimonCropp**
#skip-changelog

### CODE REVIEW

File: src/Sentry/Internal/Extensions/JsonExtensions.cs [hunk 1]
```diff
@@ -159,34 +159,35 @@ public static void Deconstruct(this JsonProperty jsonProperty, out string name,
...
+            if (dic is null)
             {
-                writer.WriteStartObject();
+                writer.WriteNullValue();
+                return;
+            }
+
+            writer.WriteStartObject();
+
+            var ordered = dic.OrderBy(_ => _.Key, StringComparer.Ordinal);
```
Comment (mattjohnsonpint) [inline src/Sentry/Internal/Extensions/JsonExtensions.cs:170]: We probably have some `OrderBy` or `Sort` in existing tests that can be removed now.

File: test/Sentry.DiagnosticSource.IntegrationTests/SqlListenerTests.RecordsEf.Core3_1.verified.txt [hunk 2]
```diff
@@ -1,4 +1,4 @@
-[
+﻿[
```
Comment (bitsandfoxes) [inline test/Sentry.DiagnosticSource.IntegrationTests/SqlListenerTests.RecordsEf.Core3_1.verified.txt:1]: I've seen this a couple of times now, what's the diff here?

File: test/Sentry.Tests/BaggageHeaderTests.cs [hunk 3]
```diff
@@ -157,7 +157,7 @@ public void BaggageHeader_Merge()
 
     private static SettingsTask VerifyHeader(BaggageHeader header)
     {
-        return Verify(header.Members)
+        return Verifier.Verify(header.Members)
```
Comment (bitsandfoxes) [inline test/Sentry.Tests/BaggageHeaderTests.cs:160]: `Verify` was just a global static method pre `2.*?

File: test/Sentry.Tests/Protocol/MeasurementTests.Transaction_Serializes_Measurements.verified.txt [hunk 4]
```diff
@@ -1,60 +1,59 @@
...
+  platform: csharp,
+  transaction: name,
+  transaction_info: {
+    source: custom
+  },
+  start_timestamp: DateTimeOffset_1,
   contexts: {
     trace: {
-      op: operation,
-      trace_id: Guid_1,
```
Comment (bitsandfoxes) [inline test/Sentry.Tests/Protocol/MeasurementTests.Transaction_Serializes_Measurements.verified.txt:5]: This one is actually different and not just a change in order. I'm not sure if that's relevant tho.

**Review by bitsandfoxes — APPROVED** [review]

**Review by mattjohnsonpint — APPROVED** [review]

File: test/Sentry.DiagnosticSource.IntegrationTests/SqlListenerTests.RecordsEf.Core3_1.verified.txt (hunk 2 above)
Comment (mattjohnsonpint) [reply test/Sentry.DiagnosticSource.IntegrationTests/SqlListenerTests.RecordsEf.Core3_1.verified.txt:1]: There used to be a [UTF-8 BOM](https://en.wikipedia.org/wiki/Byte_order_mark) at the beginning of the file, and it was deleted during merge.  Why, I'm not sure - but it doesn't affect anything.

Also, GitHub used to show a BOM with a red dot, and for some reason it doesn't anymore.  Can't find any setting to show them or any announcement that they changed something... odd.

File: test/Sentry.Tests/Protocol/MeasurementTests.Transaction_Serializes_Measurements.verified.txt (hunk 4 above)
Comment (mattjohnsonpint) [reply test/Sentry.Tests/Protocol/MeasurementTests.Transaction_Serializes_Measurements.verified.txt:5]: I think it was a bugfix in Verify.  They are indeed different guids.  It doesn't really matter here though - the main thing this one is testing is the measurements below.

Good eye though! 👍

File: test/Sentry.Tests/Protocol/SentryEventTests.SerializeObject_AllPropertiesSetToNonDefault_SerializesValidObject.verified.txt [hunk 5]
```diff
@@ -1,93 +1,92 @@
...
-      type: type
-    }
-  ],
-  contexts: {
-    .NET Framework: {
-      .NET Framework: "v2.0.50727", "v3.0", "v3.5",
-      .NET Framework Client: "v4.8", "v4.0.0.0",
-      .NET Framework Full: "v4.8"
-    },
-    context_key: context_value
```
Comment (mattjohnsonpint) [inline test/Sentry.Tests/Protocol/SentryEventTests.SerializeObject_AllPropertiesSetToNonDefault_SerializesValidObject.verified.txt:24]: @SimonCropp - FYI, I couldn't get the tests to all pass until I removed the extra `context_key` row.  No matter what I did, the sort-order was different between framework versions.  .NET 6 would sort `context_key` first, but .NET Core 3.1 and .NET 4.8 would sort `.NET Framework` first.

I think this is the same issue with Verify I was mentioning with my OSS projects.  The dictionary sort is likely using `StringComparer.InvariantCulture`, where it should probably be using `StringComparer.Ordinal`. I'm not sure if that's in Verify's code, or in a dependency.

File: test/Sentry.DiagnosticSource.IntegrationTests/SqlListenerTests.RecordsEf.Core3_1.verified.txt (hunk 2 above)
Comment (SimonCropp) [reply test/Sentry.DiagnosticSource.IntegrationTests/SqlListenerTests.RecordsEf.Core3_1.verified.txt:1]: why was the bom deleted? Verify explicitly write a bom

File: test/Sentry.Tests/Protocol/SentryEventTests.SerializeObject_AllPropertiesSetToNonDefault_SerializesValidObject.verified.txt (hunk 5 above)
Comment (SimonCropp) [reply test/Sentry.Tests/Protocol/SentryEventTests.SerializeObject_AllPropertiesSetToNonDefault_SerializesValidObject.verified.txt:24]: it is in verify. can u share a test that repos it?

Comment (mattjohnsonpint) [reply test/Sentry.Tests/Protocol/SentryEventTests.SerializeObject_AllPropertiesSetToNonDefault_SerializesValidObject.verified.txt:24]: A bit more testing, and I believe the problem isn't with Verify but with how we serialize the `Contexts` object in Sentry.  Fixed with #2031.

**Pull request merged**